core/src/main/java/org/infinispan/stream/impl/LocalStreamManagerImpl.java

```diff
@@ -179,6 +182,9 @@ public void dataRehashed(DataRehashedEvent<K, V> event) {
            Set<Integer> segments, Set<K> keysToInclude, Set<K> keysToExclude) {
       Stream<CacheEntry<K, V>> stream = (parallelStream ? cacheEntrySet.parallelStream() : cacheEntrySet.stream())
               .filterKeys(keysToInclude).filterKeySegments(segments);
+      if (cacheMode.isScattered()) {
+         stream = stream.filter(entry -> entry.getValue() != null);
```

Comment:
We could use `InternalCacheEntry::isNull`, but that would need to fix implementation in `AbstractInternalCacheEntry.isNull`. I'll check that eventually but let's leave that modification out of scope here. I plan to isolate a tombstone-related PR for the versioned history.